src/utils/tracked-controls.js

```diff
@@ -46,41 +46,81 @@ module.exports.checkControllerPresentAndSetup = function (component, idPrefix, q
-  var index = 0;
-  var isPrefixMatch;
-  var isPresent = false;
   var sceneEl = component.el.sceneEl;
   var trackedControlsSystem;
+  var targetControllerNumber;
+
+  if (!idPrefix) { return false; }
+
+  if (queryObject.hand) {
+    // This is only used in the case where the gamepads themselves are not handed
+    targetControllerNumber = (queryObject.hand === DEFAULT_HANDEDNESS) ? 0 : 1;
+  } else {
+    targetControllerNumber = queryObject.index || 0;
+  }
 
   trackedControlsSystem = sceneEl && sceneEl.systems['tracked-controls'];
   if (!trackedControlsSystem) { return false; }
 
   gamepads = trackedControlsSystem.controllers;
   if (!gamepads.length) { return false; }
 
-  for (i = 0; i < gamepads.length; ++i) {
-    gamepad = gamepads[i];
-    isPrefixMatch = (!idPrefix || idPrefix === '' || gamepad.id.indexOf(idPrefix) === 0);
-    isPresent = isPrefixMatch;
-    if (isPresent && queryObject.hand) {
-      isPresent = (gamepad.hand || DEFAULT_HANDEDNESS) === queryObject.hand;
+  return !!findMatchingController(gamepads, null, idPrefix, queryObject.hand, targetControllerNumber);
+}
+
+module.exports.isControllerPresent = isControllerPresent;
+
+/**
+ * Walk through the given controllers to find any where the device ID equals filterIdExact, or startWith filterIdPrefix.
+ * A controller where this considered true is considered a 'match'.
+ *
+ * For each matching controller:
+ *   If filterHand is set, and the controller:
+ *     is handed, we further verify that controller.hand equals filterHand.
+ *     is unhanded (controller.hand is ''), we skip until we have found a number of matching controllers that equals targetControllerNumber
+ *   If filterHand is not set, we skip until we have found the nth matching controller, where n equals targetControllerNumber
+ *
+ * The method should be called with one of: [filterIdExact, filterIdPrefix] AND one or both of: [filterHand, targetControllerNumber]
+ *
+ * @param {object} controllers - Array of gamepads to search
+ * @param {string} filterIdExact - If set, used to find controllers with id === this value
+ * @param {string} filterIdPrefix - If set, used to find controllers with id startsWith this value
+ * @param {object} filterHand - If set, further filters controllers with matching 'hand' property
+ * @param {object} targetControllerNumber - Find the nth matching controller, where n equals targetControllerNumber. defaults to 0.
+ */
+function findMatchingController (controllers, filterIdExact, filterIdPrefix, filterHand, targetControllerNumber) {
+  var controller;
+  var i;
+  var matchingControllerOccurence = 0;
+  targetControllerNumber = targetControllerNumber || 0;
+
+  for (i = 0; i < controllers.length; i++) {
+    controller = controllers[i];
+    // Determine if the controller ID matches our criteria
+    if (filterIdPrefix && controller.id.indexOf(filterIdPrefix) === -1) continue;
+    if (!filterIdPrefix && controller.id !== filterIdExact) continue;
+
+    if (filterHand) {
```

Comment:
Maybe simpler like this? And maybe more accurate if there are several controllers with the same handedness. I think it cannot happen today with the available hardware but who knows if one day 2 players can play in the same machine with two pairs of controllers
````javascript
// If the hand filter and controller handedness are defined we compare them.
if (filterHand && controller.hand && filterHand !== controller.hand) { continue; }
````

Reply:
Maybe also more consistent with the negative statements above

Reply:
Good point re: multiple pairs, will make appropriate changes and add tests.

Reply:
from a little experience - undefined controller hand means none, but undefined filterHand means don't care, which is a completely different thing.  See #2583

Reply:
you want to be able to specify a filter for no-handedness (vive-tracker), as well as for specific handedness (various handed controllers), as well as don't care (don't care whether the 3DOF controller is left/right/none, just give me the pose)

Reply:
It looks like your first and third scenarios can be handled in the same way, by simply passing in filterHand = undefined, and optionally using the targetControllerNumber to get a controller at specific index

Reply:
no - filterHand undefined (default) means "I don't care whether controller.hand exists, much less what its value is".  That cannot be used for the first case (no-handedness), where the test must be performed, and must only succeed when controller.hand is undefined.

Reply:
@machenmusik not sure I fully understand. Can you suggest an alternative if statement that covers the case you're referring?

Reply:
https://github.com/aframevr/aframe/pull/2583/files#r136465173 is a bit ugly but working.
- instead of relying on schema hand undefined (which is a little ambiguous about no-handedness vs. don't-care as discussed here), use explicit strings `any-or-none` and `none` to specify hand filter

Reply:
I think the utils function change is https://github.com/aframevr/aframe/pull/2583/files#diff-24b039fab259237004f42c5bf3856cdfR69 -- use DEFAULT_HANDEDNESS to replace controller hand of undefined with an actual string to match

Reply:
but actually I think `DEFAULT_HANDEDNESS` was supposed to be `none` there

Reply:
We can revisit this later for tracker support.

Reply:
generally though, something like:
```
// Only try to match filterHand if a filter value is specified.
if (filterHand) {
  if ((controller.hand ? controller.hand : DEFAULT_HANDEDNESS) !== filterHand) { continue; }
}
```

Reply:
If the filter does not reject based on hand value, I believe we want to fall through against controller index check (if any) so you can match against the second set of left and right hand controllers, or the second or third trackers.